src/components/NewCommentForm.tsx

```
@@ -59,13 +63,19 @@ export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
 
     setSubmitting(true);
 
-    // it is very easy to forget about `await` keyword
-    await onSubmit({ name, email, body });
-
-    // and the spinner will disappear immediately
-    setSubmitting(false);
-    setValues(current => ({ ...current, body: '' }));
-    // We keep the entered name and email
+    const newComment = {
+      name,
+      email,
+      body,
+      postId: post.id,
+    };
+
+    try {
+      await dispatch(commentsActions.initAdd(newComment));
+      setValues(current => ({ ...current, body: '' }));
+    } finally {
+      setSubmitting(false);
+    }
```

Error handling / user feedback: currently you `await dispatch(commentsActions.initAdd(newComment))` inside a try/finally but you don't catch errors. Prefer using `.unwrap()` so rejections throw and can be caught, and show some feedback to the user in the catch block. Example pattern:

```ts
try {
  await dispatch(commentsActions.initAdd(newComment)).unwrap();
  setValues(current => ({ ...current, body: '' }));
} catch (err) {
  // set local error state or notify user
} finally {
  setSubmitting(false);
}
```

This prevents swallowed errors and lets you display a friendly message or restore UI state on failure.